Gems/PhysX/Code/Source/PrismaticJointComponent.cpp

```diff
@@ -84,4 +87,83 @@ namespace PhysX
             m_jointSceneOwner = leadFollowerInfo.m_followerBody->m_sceneOwner;
         }
     }
+
+    // AZ::Component
+    void PrismaticJointComponent::Activate() {
+        JointComponent::Activate();
+        PhysX::JointInterfaceRequestBus::Handler::BusConnect(this->GetEntityId());
+    };
+
+    void PrismaticJointComponent::Deactivate() {
+        JointComponent::Deactivate();
+        PhysX::JointInterfaceRequestBus::Handler::BusDisconnect();
+    };
+
+    physx::PxD6Joint* GetPhysXD6Joint(const AzPhysics::JointHandle& joint_handle,
+                                                        const AzPhysics::SceneHandle& joint_scene_owner)
+    {
+        auto* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get();
+        const auto joint = sceneInterface->GetJointFromHandle(joint_handle,joint_scene_owner);
+        physx::PxJoint* native = static_cast<physx::PxJoint*>(joint->GetNativePointer());
+        if (native && native->is<physx::PxD6Joint>())
+        {
+            physx::PxD6Joint *native_d6 = static_cast<physx::PxD6Joint*>(joint->GetNativePointer());
+            return native_d6;
+        }
+        return nullptr;
+    }
+
+    float PrismaticJointComponent::GetPosition(){
+        auto* native = GetPhysXD6Joint(m_jointSceneOwner,m_jointHandle);
+        if (native){
+            return native->getRelativeTransform().p.x;
+        }
+        return 0.f;
+    };
+
+    float PrismaticJointComponent::GetVelocity() {
+        auto* native = GetPhysXD6Joint(m_jointSceneOwner,m_jointHandle);
+        if (native){
+            return native->getRelativeLinearVelocity().x;
```

Comment:
Why only X component of the velocity?

Reply:
Added explanation to code.

Reply:
````
            // Undelying PhysX joint is D6, but it simulates PhysXPrismatic joint.
            // The D6 joint has only X-axis unlocked, so report only X velocity.
````

Thanks for the comment, it makes it super clear now!